### PR TITLE
feat(query): "Callback Object With Incorrect Ref" for OpenAPI (#3076)

### LINKED ISSUE #3076: Add "Callback Object With Incorrect Ref" query for OpenAPI
### Platform
OpenAPI

### Description
Callback Object reference must always point to '#components/callbacks'

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares rafaela.soares@checkmarx.com

Closes #3076

Proposed Changes
- Added "Callback Object Incorrect Ref" query for OpenAPI. It checks if callback object reference does not point to '#components/callbacks'

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/libraries/openapi/library.rego [hunk 1]
```diff
@@ -16,3 +16,15 @@ improperly_defined(params, value) {
...
+incorrect_ref(ref, object) {
+	references := {
+		"schema": "#/components/schemas/",
+		"responses": "#/components/responses/",
+		"requestBody": "#/components/requestBodies/",
+		"links": "#/components/links/",
+		"callbacks": "#/components/callbacks/",
+	}
+
+	regex.match(references[object], ref) == false
```
Comment (felipe-avelar) [inline assets/libraries/openapi/library.rego:29]: This is going to be very useful, nice addition :)

File: assets/queries/openAPI/callback_object_incorrect_ref/query.rego [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+	call_back := doc.paths[n][oper].callbacks[c]
+
+	openapi_lib.incorrect_ref(call_back["$ref"], "callbacks")
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("openapi.paths.%s.%s.responses.%s.$ref={{%s}}", [n, oper, c, call_back["$ref"]]),
```
Comment (felipe-avelar) [inline assets/queries/openAPI/callback_object_incorrect_ref/query.rego:14]: Shouldn't be somenthing like `openapi.paths.{{%s}}.%s.responses.%s.$ref`?

**Review by felipe-avelar — COMMENTED** [review]
Please, check comments

**Review by joaoReigota1 — DISMISSED** [review]
LGTM

**Comment by rogeriopeixotocx** [comment]
@rafaela-soares please solve conflicts

File: assets/queries/openAPI/callback_object_incorrect_ref/query.rego (hunk 2 above)
Comment (rafaela-soares) [reply assets/queries/openAPI/callback_object_incorrect_ref/query.rego:14]: My bad. I thought it was necessary to specify but it is not. Thank you!

**Review by rogeriopeixotocx — APPROVED** [review]

**Pull request merged**